hosts/testagent/configuration.nix

```diff
@@ -0,0 +1,115 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Edit this configuration file to define what should be installed on
+# your system.  Help is available in the configuration.nix(5) man page
+# and in the NixOS manual (accessible by running ‘nixos-help’).
+{
+  self,
+  inputs,
+  ...
+}: {
+  imports =
+    [
+      # Include the results of the hardware scan.
+      ./testagent-hardware-configuration.nix
+    ]
+    ++ (with inputs; [
+      sops-nix.nixosModules.sops
+      disko.nixosModules.disko
+    ])
+    ++ (with self.nixosModules; [
+      common
+      service-openssh
+      user-tervis
+      user-vjuntunen
+      user-flokli
+    ]);
+
+  # Bootloader.
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  networking.hostName = "testagent"; # Define your hostname.
+  # networking.wireless.enable = true;  # Enables wireless support via wpa_supplicant.
+
+  # Configure network proxy if necessary
+  # networking.proxy.default = "http://user:password@proxy:port/";
+  # networking.proxy.noProxy = "127.0.0.1,localhost,internal.domain";
+
+  # Enable networking
+  networking.networkmanager.enable = true;
+
+  # Select internationalisation properties.
+  i18n.defaultLocale = "en_US.UTF-8";
+
+  i18n.extraLocaleSettings = {
+    LC_ADDRESS = "fi_FI.UTF-8";
+    LC_IDENTIFICATION = "fi_FI.UTF-8";
+    LC_MEASUREMENT = "fi_FI.UTF-8";
+    LC_MONETARY = "fi_FI.UTF-8";
+    LC_NAME = "fi_FI.UTF-8";
+    LC_NUMERIC = "fi_FI.UTF-8";
+    LC_PAPER = "fi_FI.UTF-8";
+    LC_TELEPHONE = "fi_FI.UTF-8";
+    LC_TIME = "fi_FI.UTF-8";
+  };
+
+  # Disable suspend and hibernate - systemd settings
+  services.logind.extraConfig = ''
+    HandleSuspendKey=ignore
+    HandleLidSwitch=ignore
+    HandleLidSwitchDocked=ignore
+    HandleHibernateKey=ignore
+  '';
+
+  # Ensure the system does not automatically suspend or hibernate
+  # This is an additional measure to above, can be adjusted as needed
+  services.upower.enable = false;
+
+  # Disable the GNOME3/GDM auto-suspend feature that cannot be disabled in GUI!
+  # If no user is logged in, the machine will power down after 20 minutes.
+  systemd.targets.sleep.enable = false;
+  systemd.targets.suspend.enable = false;
+  systemd.targets.hibernate.enable = false;
+  systemd.targets.hybrid-sleep.enable = false;
+
+  # Disable the X11 windowing system.
+  services.xserver.enable = false;
```

Comment:
This shouldn't be enabled as long as we don't enable any desktop environment, can be removed.